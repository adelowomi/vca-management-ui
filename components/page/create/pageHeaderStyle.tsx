import * as React from 'react';

export interface PageHeaderStyleProps {
  onButtonClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  state: any;
}

export function PageHeaderStyle({
  onButtonClick,
  state,
}: PageHeaderStyleProps) {
  return (
    <div className="header-button-group flex justify-start space-x-3 ">
      <button
        onClick={onButtonClick}
        value="headerTypeOne"
        data-headertype="headerTypeOne"
        type="button"
        className={`
      ${
        state && state.headerType === 'headerTypeOne'
          ? 'bg-indigo-200 text-indigo-600 '
          : ''
      }
      inline-flex items-center px-3 py-2 border shadow-sm text-base font-medium rounded-md text-gray-500 focus:outline-none 
     
      `}
      >
        Header Type 1
      </button>
      <button
        value="headerTypeTwo"
        data-headertype="headerTypeTwo"
        onClick={onButtonClick}
        type="button"
        className={`
      ${
        state && state.headerType === 'headerTypeTwo'
          ? 'bg-indigo-200 text-indigo-600 '
          : ''
      }
      inline-flex items-center px-3 py-2 border shadow-sm text-base font-medium rounded-md text-gray-500  focus:outline-none 
     
      `}
      >
        Header Type 2
      </button>
      <button
        value="headerTypeThree"
        data-headertype="headerTypeThree"
        onClick={onButtonClick}
        type="button"
        className={`
      ${
        state && state.headerType === 'headerTypeThree'
          ? 'bg-indigo-200 text-indigo-600 '
          : ''
      }
      inline-flex items-center px-3 py-2 border shadow-sm text-base font-medium rounded-md text-gray-500  focus:outline-none 
     
      `}
      >
        Header Type 3
      </button>
      <button
        value="headerTypeFour"
        data-headertype="headerTypeFour"
        onClick={onButtonClick}
        type="button"
        className={`
      ${
        state && state.headerType === 'headerTypeFour'
          ? 'bg-indigo-200 text-indigo-600 '
          : ''
      }
      inline-flex items-center px-3 py-2 border shadow-sm text-base font-medium rounded-md text-gray-500  focus:outline-none 
     
      `}
      >
        Header Type 4
      </button>
    </div>
  );
}
