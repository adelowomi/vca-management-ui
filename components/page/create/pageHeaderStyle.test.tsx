import React from 'react';

import { fireEvent, render, screen } from '../../../test/testUtils';
import { PageHeaderStyle } from './pageHeaderStyle';

describe('PageHeaderStyle', () => {
  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    const { asFragment } = render(
      <PageHeaderStyle onButtonClick={mockOnButtonClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    render(<PageHeaderStyle onButtonClick={mockOnButtonClick} />);
    fireEvent.click(screen.getByText('Header Type 1'));
    expect(mockOnButtonClick).toHaveBeenCalled();
  });
});
