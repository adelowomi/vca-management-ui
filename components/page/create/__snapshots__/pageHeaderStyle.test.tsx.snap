// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeaderStyle should render PageHeaderStyle snapshot 1`] = `
<DocumentFragment>
  <div
    class="header-button-group flex justify-start space-x-3 "
  >
    <button
      class="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      type="button"
      value="headerTypeOne"
    >
      Header Type 1
    </button>
    <button
      class="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      type="button"
      value="headerTypeTwo"
    >
      Header Type 2
    </button>
    <button
      class="inline-flex items-center px-3 py-2 border  shadow-sm text-base font-medium rounded-md text-indigo-700 bg-indigo-200 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      type="button"
      value="headerTypeThree"
    >
      Header Type 3
    </button>
    <button
      class="inline-flex items-center px-3 py-2 border border-gray-300 shadow-sm text-base font-medium rounded-md text-gray-500 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
      type="button"
      value="headerTypeFour"
    >
      Header Type 4
    </button>
  </div>
</DocumentFragment>
`;
